web/middleware/etag.go

```diff
@@ -0,0 +1,49 @@
+package middleware
+
+import (
+	"crypto/md5"
+	"net/http"
+	"fmt"
+)
+
+type etagResponseWriter struct {
+	status int
+	w http.ResponseWriter
+	r *http.Request
+}
+
+func (e *etagResponseWriter) Header() http.Header {
+	return e.w.Header()
+}
+
+func (e *etagResponseWriter) WriteHeader(s int) {
+	e.w.WriteHeader(s)
+	e.status = s
+}
+
+func (e *etagResponseWriter) Write(b []byte) (int, error) {
+	eTag := fmt.Sprintf("%x", md5.Sum(b))
```

Comment:
This will only work if you can `Write` out your message in a single call. Unfortunately many uses of (e.g.,) `io.Copy` will copy large bodies in chunks, so this will produce incorrect results